runtime/lib/ttmetal/command_queue.cpp

```diff
@@ -370,9 +418,19 @@ static ::tt::tt_metal::CircularBufferConfig createCircularBufferConfig(
       cbRef->desc()->memory_desc()->size() * cbRef->desc()->num_buffers();
   ::tt::DataFormat dataFormat =
       toDataFormat(cbRef->desc()->memory_desc()->data_type());
-  LOG_ASSERT(cbRef->tensor_ref());
-  LOG_ASSERT(cbRef->tensor_ref()->address() == cbRef->address(),
-             "Address mismatch between tensor ref and cb ref");
+
+  if (!cbRef->tensor_ref()) {
+    assert(cbRef->address() >=
+           buffers.at(CQExecutor::kReservedBufferId)->address());
+    // TODO specifying address to CB must be done through shadow buffer.
+    // However, in this case we're seeing weird behavior w.r.t. to results when
+    // using shadow buffer. This needs to be investigated further.
+    return CircularBufferConfig(totalSize,
+                                {{cbRef->desc()->port(), dataFormat}})
+        // *buffers.at(CQExecutor::kReservedBufferId))
```

Comment:
CB can be allocated in 2 ways --
1. No buffer - metal will allocate it at arbitrary address.
2. With (shadow) buffer - we can control address and lifetime of the buffer. 

I observed incorrect results of math op when using the second option which is preferred to us because we can control allocations. I did not find the cause of the issue, maybe I programmed the shadow buffer incorrectly in func `createReservedL1Buffer`. 

By incorrect results I mean there are weird numbers and weird patterns, for example, doing reduce W only first 16 rows show non-zero result at the left most column, while rest of the rows are all zeros. It may have something to do with tilizing, as I've seen similar weird patterns before when I tested tilizing feature.

@nsmithtt  If you have a hunch about this please share.

Reply:
Yeah this sounds most likely to be a tiling issue.

Reply:
Just to be clear, I do not guarantee for numerical correctness as of now (even without this tiling issue), as I did not test vs golden.